evennia/utils/verb_conjugation/pronouns.py

```diff
@@ -350,44 +381,42 @@ def pronoun_to_viewpoints(
-    pronoun_types = PRONOUN_MAPPING[pronoun_lower]
-    # this has one or more pronoun-types
-    if len(pronoun_types) == 1:
-        pronoun_type, viewpoints = next(iter(pronoun_types.items()))
-    elif pronoun_type in pronoun_types:
-        viewpoints = pronoun_types[pronoun_type]
-    elif DEFAULT_PRONOUN_TYPE in pronoun_types:
-        pronoun_type = DEFAULT_PRONOUN_TYPE
-        viewpoints = pronoun_types[pronoun_type]
-    else:
-        # not enough info - grab the first of the mappings
-        pronoun_type, viewpoints = next(iter(pronoun_types.items()))
-
-    # we have one or more viewpoints at this point
-    if len(viewpoints) == 1:
-        viewpoint, genders = next(iter(viewpoints.items()))
-    elif viewpoint in viewpoints:
-        genders = viewpoints[viewpoint]
-    elif DEFAULT_VIEWPOINT in viewpoints:
-        viewpoint = DEFAULT_VIEWPOINT
-        genders = viewpoints[viewpoint]
+    # check if pronoun maps to multiple options and differentiate
+    # but don't allow invalid differentiators
+    if is_iter(source_type):
+        pronoun_type = pronoun_type if pronoun_type in source_type else source_type[0]
     else:
-        # not enough info - grab first of mappings
-        viewpoint, genders = next(iter(viewpoints.items()))
-
-    # we have one or more possible genders (including plural forms)
-    if len(genders) == 1:
-        gender, mapped_pronoun = next(iter(genders.items()))
-    elif gender in genders:
-        mapped_pronoun = genders[gender]
-    elif DEFAULT_GENDER in genders:
-        gender = DEFAULT_GENDER
-        mapped_pronoun = genders[gender]
+        pronoun_type = source_type
+    target_viewpoint = VIEWPOINT_CONVERSION[source_viewpoint]
+    if is_iter(target_viewpoint):
+        viewpoint = viewpoint if viewpoint in target_viewpoint else target_viewpoint[0]
     else:
-        # not enough info - grab first mapping
-        gender, mapped_pronoun = next(iter(genders.items()))
+        viewpoint = target_viewpoint
 
+    # special handling for the royal "we"
+    if is_iter(source_gender):
+      gender_opts = list(source_gender)
+    else:
+      gender_opts = [source_gender]
+    if viewpoint == "1st person":
+        # make sure plural is always an option when converting to 1st person
+        # it doesn't matter if it's in the list twice, so don't bother checking
+        gender_opts.append("plural")
+    # if the gender is still not in the extended options, fall back to source pronoun's default
+    gender = gender if gender in gender_opts else gender_opts[0]
+
+    # step down into the mapping
+    viewpoint_map = PRONOUN_MAPPING[viewpoint]
+    
+    if pronoun_type in viewpoint_map:
+        pronouns = viewpoint_map[pronoun_type]
+    else:
+        pronouns = viewpoint_map[DEFAULT_PRONOUN_TYPE]
+    if gender in pronouns:
+        mapped_pronoun = pronouns[gender]
+    else:
+        mapped_pronoun = pronouns[DEFAULT_GENDER]
```

Comment:
It's clearer and more succinct to write these as

```python
pronouns = viewpoint_map.get(pronoun_type, DEFAULT_PRONOUN_TYPE)
mapped_pronoun = pronouns.get(gender, DEFAULT_GENDER)
```

Reply:
Oh, good catch!

Reply:
Would this be better than the original?
```py
    viewpoint_map = PRONOUN_MAPPING[viewpoint]

    pronouns = viewpoint_map.get(pronoun_type)
    if not pronouns:
        pronouns = viewpoint_map[DEFAULT_PRONOUN_TYPE]
    
    mapped_pronoun = pronouns.get(gender)
    if not mapped_pronoun:
        mapped_pronoun = pronouns[DEFAULT_GENDER]
```

Reply:
Ah, never mind that, I figured out how to trim it down further

Reply:
Oh, you are right, I was too hasty. :) Yeah, it'd need to be `pronouns.get(gender, pronouns[DEFAULT_GENDER])`